### ISSUE TITLE
Github View/Edit links broken if defaultContentLanguageInSubdir = false

### ISSUE THREAD

**Issue by fekete-robert** [google/docsy#1420]
Hi, I am working on a site that uses Docsy 0.6.0 and might have hit a bug related to the following settings:

```
contentDir = "content"
defaultContentLanguage = "en"
defaultContentLanguageInSubdir = false
```

The site is only available in English, so we don't use the content/en/ subdir. The site builds fine, but the github links "View/edit this page" are broken, because they do contain the "/en/" part.

I wanted to create a PR to omit the language from https://github.com/google/docsy/blob/main/layouts/partials/page-meta-links.html if "defaultContentLanguageInSubdir = false", but since this parameter is not under the [params] tag of the config file, it's not accessible from the partial (unless I'm missing something).

Does anyone have a solution or workaround to the problem? It's not a big issue, if there is no neat solution I'll just hide the links. An ugly workaround would be to duplicate the "defaultContentLanguageInSubdir = false" under [params] as well, but that's a really ugly hack.

The site is https://kube-logging.github.io/docs/ , its source is available at https://github.com/kube-logging/kube-logging.github.io/blob/master/config/_default/config.toml

Thanks!
Robert

**Comment by chalin** [comment]
I have a similar setup for another project, and there are no issues with the GH page links. I'll investigate further and get back to you soon (probably later today).

**Comment by chalin** [comment]
I haven't double checked yet, but off the top of my head what I recall is that you'll need to replace the `contentDir` by a mount point because we're not supposed to mix mount points (which are required when using Hugo modules) with the old `*Dir` config parameters.

From https://gohugo.io/hugo-modules/configuration/#module-config-mounts (emphasis mine):

> When the `mounts` config was introduced in Hugo 0.56.0, we were careful to preserve the existing `contentDir`, `staticDir`, and similar configuration to make sure all existing sites just continued to work. <ins>**But you should not have both**: **if you add a `mounts` section you should remove the old `contentDir`**</ins>, `staticDir`, etc. settings.

Let me know if this helps.

**Comment by fekete-robert** [comment]
We don't use module mounts for this project, so it's not mixed. I've played around them a bit but didn't seem to solve the issue. I'll override the partial locally in our project to fix the links for now.

**Comment by LisaFC** [comment]
It would be great if @chalin's workaround works because this is an old, old problem - as you discovered, you can't actually find defaultContentLanguageInSubdir via params in the partial, so there have been a variety of hacky solutions from various Docsy users, including overriding the entire partial.

**Comment by fekete-robert** [comment]
@LisaFC I couldn't get it work, but I have found another one that might be good enough to create an "official" workaround.

You can access the `defaultContentLanguage = "en"` and `defaultContentLanguageInSubdir = false` config parameters from the partials if you set them to cascade in the config file, like this:

```
[cascade]
defaultContentLanguage = "en"
defaultContentLanguageInSubdir = false
```

If this is set in the config file, then something like this works in the partial (I don't manage multilanguage sites so I don't know what the proper conditions would be):

```
{{ if and (eq ($.Param "defaultContentLanguage") (.Site.Language.Lang)) ($.Param "defaultContentLanguageInSubdir") -}}
    <!-- If defaultContentLanguageInSubdir is true and defaultContentLanguage is the current language -->
    {{ $gh_repo_path = printf "%s/content/%s/%s" $gh_branch ($.Site.Language.Lang) $pathFormatted -}}
    {{ else if and (eq ($.Param "defaultContentLanguage") (.Site.Language.Lang)) (not ($.Param "defaultContentLanguageInSubdir")) -}}
    <!-- If defaultContentLanguageInSubdir is false and defaultContentLanguage is the current language -->
    {{ $gh_repo_path = printf "%s/content/%s" $gh_branch $pathFormatted -}}
```

What do you think?

**Comment by chalin** [comment]
Nice site you have there!

> We don't use module mounts for this project, so it's not mixed

If you're using Docsy, since Docsy uses mounts, then your project is using mounts too: https://github.com/google/docsy/blob/5597d435dc74ce68240e0c3871addf24567493b0/config.yaml#L24-L46

> I've played around them a bit but didn't seem to solve the issue.

Ok. I'll take a look as soon as I can (quite busy atm). I'm starting to feel that this _is_ a bug in the page-link rendering.

In the meantime, you should be able to use the [path_base_for_github_subdir](https://www.docsy.dev/docs/adding-content/repository-links/#path_base_for_github_subdir-optional) feature (with `from` and `to` fields) to get around the bug (to remove the `/en` from link paths).

**Comment by bdalpe** [comment]
I was able to use the workaround provided by @chalin. Sample toml config:

```toml
[params.path_base_for_github_subdir]
# change content/<lang>/... to content/...
from = "content/(.*?)/(.*?)"
to = "content/$2"
```

**Comment by chalin** [comment]
@fekete-robert - can you confirm whether the following would have fixed the issue?

- #1744

**Comment by fekete-robert** [comment]
Yes, it seems to work. Thanks!

**Comment by chalin** [comment]
#1744 is the main fix in 0.9.0.
Thanks for confirming that it is working now for you.

Closed by #1744
